src/Mod/TemplatePyMod/FeaturePython.py

```diff
@@ -144,6 +144,7 @@ def makeLine():
 	Line(a)
 	#ViewProviderLine(a.ViewObject)
 	a.ViewObject.Proxy=0 # just set it to something different from None
+	FreeCAD.ActiveDocument.recompute()

```

Comment:
Do not add the recomputation inside the make function. The recomputation must be outside the function, once the objects are already created.
```py
class Line:
    ...

def make_line():
    ...

obj = make_line()
FreeCAD.ActiveDocument.recompute()
```

Reply:
The problem with that is that this is example code and if you don't recompute inside the 'make' function then there is nowhere else to do it.  Your second-last line (`obj = make_line()`) is also missing, I assume for some good reason; this is an example showing how to do it, not code that we ever expect to be called.

Without putting the recompute in there somewhere, it's not at all obvious that you _need_ to recompute before the 3D view will show anything (trust me, I wasted a day and a half figuring this out).  Would a comment there saying, "Consider carefully when to trigger recomputation as doing so many times is expensive and might affect the UI responsivity" address your concerns?

Reply:
Well, it's an example showing you how to *define* feature classes and viewprovider classes, so we should define them like the examples show, and not add additional recomputation steps. That wouldn't be good examples.

To clarify the usage, you may add a module docstring with some information.
```py
"""
Typical usage

import FeaturePython
obj1 = FeaturePython.make_line()
obj2 = FeaturePython.make_circle()
obj3 = FeaturePython.make_shape()

# We need to recompute the document to calculate the shapes of the new objects
App.ActiveDocument.recompute()
"""
```

This is what other examples demonstrate, for example, [Draft_Line](https://wiki.freecadweb.org/Draft_Line) or [Draft_ShapeString](https://wiki.freecadweb.org/Draft_ShapeString).